Week-23/todo-list/Controller/TodoController.js

```diff
@@ -0,0 +1,71 @@
+const Todo = require("../Model/TodoModel");
+exports.postTodo = async (req, res) => {
+  try {
+    const { todo } = req.body;
+    const newTodo = new Todo({ todo });
+    await newTodo.save();
+    res.status(201).json(newTodo);
+  } catch (error) {
+    console.error("Error inserting todo:", error);
+    res.status(500).json({ error: "Internal Server Error" });
+  }
+};
+//get aallTodos
+exports.getTodos = async (req, res) => {
+  try {
+    const result = await Todo.find();
+    if(result.length==0){
+        res.status(404).json({msg:'No Todos Found'})
+    }
+    res.status(200).json(result);
+  } catch (error) {
+    console.log("error", error);
+  }
+};
+
+//get todo by id
+exports.getTodoByID = async (req, res) => {
+  const id = req.params.id;
+  try {
+    const result = await Todo.findById(id);
+    console.log("result id", result);
+    res.status(201).json(result);
```

Comment:
```suggestion
    res.status(200).json(result);
```
201 is only used as created status code. Here we are fetching records which if successful should respond with OK status. Also, rather than hardcoding status ids, use http status npm package or built-in ways to send enums instead of status ids, so that code is more readable and easier to understand.